assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,307 @@
+
+  // Recently used font options, clean up on refresh
+  const recentUsedFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? recentUsedFontSlugs
+            .map((slug) => fonts.find((font) => font.slug === slug))
+            .filter(({ name }) =>
+              name.toLowerCase().includes(searchValue.toLowerCase())
+            )
+        : recentUsedFontSlugs.map((slug) =>
+            fonts.find((font) => font.slug === slug)
+          ),
+    [searchValue, recentUsedFontSlugs, fonts]
+  );
+
+  // Font options that start with search term
+  const normalFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(({ name }) =>
+            name.toLowerCase().startsWith(searchValue.toLowerCase())
+          )
+        : fonts,
+    [searchValue, fonts]
+  );
+
+  // Font options that include search term but not start with
+  const includeSearchFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(
+            ({ name }) =>
+              name.toLowerCase().indexOf(searchValue.toLowerCase()) > 0
+          )
+        : [],
+    [searchValue, fonts]
+  );
+
+  useEffect(() => {
+    const combinedFontList = []
+      .concat(recentUsedFonts)
+      .concat(normalFonts)
+      .concat(includeSearchFonts)
+      .slice(currentIndex, currentIndex + 10)
+      .map(({ name }) => name);
+
+    if (combinedFontList.length > 0) {
+      getMenuFonts(combinedFontList).then((result) => {
+        const resultArray = result.replace(/}/g, '}},').split('},');
+        setMenuFonts(resultArray);
+      });
+    }
+  }, [
+    currentIndex,
+    getMenuFonts,
+    recentUsedFonts,
+    normalFonts,
+    includeSearchFonts,
+  ]);
+
+  useEffect(() => {
+    inputRef.current.focus();
+  });
+
+  useEffect(() => {
+    const listElement = listContainerRef.current;
+
+    const handleScroll = () => {
+      let { scrollTop } = listElement;
+      scrollTop -= 5;
+      let currentVisibleIndex = Math.floor(scrollTop / 34);
+      if (
+        recentUsedFonts.length > 0 &&
+        currentVisibleIndex > recentUsedFonts.length
+      ) {
+        scrollTop -= 10;
+        currentVisibleIndex =
+          Math.floor(scrollTop / 34) - recentUsedFonts.length;
+      }
+      if (normalFonts.length > 0 && currentVisibleIndex > normalFonts.length) {
+        scrollTop -= 10;
+      }
+      currentVisibleIndex = Math.floor(scrollTop / 34);
+      if (currentVisibleIndex < 0) {
+        currentVisibleIndex = 0;
+      }
+      if (currentVisibleIndex !== currentActiveRef.current) {
+        currentActiveRef.current = currentVisibleIndex;
+        setCurrentIndex(currentVisibleIndex);
+      }
+    };
+    listElement.addEventListener('scroll', handleScroll, { passive: false });
+
+    return () => {
+      listElement.removeEventListener('scroll', handleScroll);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This only needs to be called once on mount. And there was dependency issues with it. So disabled it.

Reply:
Can you enable and let me know what dependency issues you're seeing? Overall, this lint should never be disabled, except when you are making a custom hook.

Reply:
We have normalFonts (will be changed to otherFonts) and recentUsedFonts used. So when I disable it I got this.
`missing dependencies: normalFonts.length and recentUsedFonts. Either include them or remove dependencies`

Reply:
From the shape of this effect, it looks like you actually need all of these dependencies. Maybe you need to change how this hook is setup in general and what data it's working on? Otherwise it'd seem like you'd have to add them and remove lint disabling.

Reply:
Updated. And add scroll listener will be called once.

Reply:
You mean it will be called once per `otherFonts` and `recentUsedFonts` changes?

Reply:
Just checked. And it was my mistake. Previously there was currentIndex with setCurrentIndex which causing frequent update for that effect. Instead useRef used and including otherFonts and recentusedFonts changes will not effect. Thank you

Reply:
Just confirmed. And it will not effect! Previously currentIndex and setCurrentIndex used on handleScroll event and it caused that. So useRef used instead. Thank you